Fix Foreman returning host parameters

milliams opened the pull request:
Backport PR for #51034

milliams committed:
Fix Foreman returning host parameters 

Foreman (1.20) returns the `all_parameters` key as a list of dicts, not a dict of key-value pairs.

abadger commented on lib/ansible/plugins/inventory/foreman.py:
```diff
@@ -166,8 +166,8 @@ def _get_all_params_by_id(self, hid):
         if not ret or not isinstance(ret, MutableMapping) or not ret.get('all_parameters', False):
-            ret = {'all_parameters': [{}]}
-        return ret.get('all_parameters')[0]
+            ret = []
```
When this path is triggered, it will raise an Exception because lists do not have a get() method.  So ```return ret.get('all_parameters')``` will fail.

milliams replied:
I have cherry-picked the commits from #54333

abadger reviewed (changes requested):
I notice a regression in the new code.  Can you fix that in devel and then add the new commit to this backport?

This also needs a changelog fragment before it's merged. https://docs.ansible.com/ansible/latest/community/development_process.html?highlight=backport#backport-pull-request-process

milliams committed:
Fix for type error

The empty type here should be a dict, not a list as is has a `get` done
on it next.

(cherry picked from commit e94e80c79eab425637638205efa0eae368389dcd)

milliams commented:
@abadger thanks for the spot. Once the patch is merged to devel (#54333) I'll cherry-pick it here and add the changelog fragment.

milliams committed:
Return dict directly to avoid failing key lookup

(cherry picked from commit 545b98645d04b2d26eb95354d8edf4d8e9d4e2a2)

milliams committed:
Add changelog fragment for #54333

milliams commented:
I've made the changes and have added a changelog fragment. This is my first time backporting a fix so let me know if I've done anything wrong.

abadger commented:
Thanks :-)  Merged for the 2.7.10 release.
